utils/tagged_integer.hh

```diff
@@ -98,4 +98,19 @@ template <typename Final, typename Tag, std::integral ValueType>
     return s << x.value();
 }
 
+template <typename Final, typename Tag, std::integral ValueType>
+struct numeric_limits<utils::tagged_tagged_integer<Final, Tag, ValueType>> : public numeric_limits<ValueType> {
+    using type = utils::tagged_tagged_integer<Final, Tag, ValueType>;
```

Comment:
no need to define this. on the contrary, the reason why we can inherit from `numeric_limits<ValueType>` is that we rely on the fact that `numeric_limits<ValueType>::is_specialized` and `numeric_limits<ValueType>::is_bounded` are true. so we either need to have
```c++
using value_limits =  numeric_limits<ValueType>;
static_assert(value_limits::is_specialized && value_limits::is_bounded);
```
to make this assumption more explicit.
or we just trust that the library complies to the standard that all built-in fundamental types are bounded and the standard library specializes `numeric_limits` for all integral types.

also, i'd suggest add
```c++
static_assert(static_assert(std::numeric_limits<version_type>::is_bounded);
```

in `heart_beat_state::force_highest_possible_version_unsafe()`, before calling `std::numeric_limits<version_type>::max()`, as the caller site has this assumption on `version_type`, so it might want to make this explicit.

Reply:
> no need to define this. on the contrary, the reason why we can inherit from `numeric_limits<ValueType>` is that we rely on the fact that `numeric_limits<ValueType>::is_specialized` and `numeric_limits<ValueType>::is_bounded` are true. so we either need to have
> 
> ```c++
> using value_limits =  numeric_limits<ValueType>;
> static_assert(value_limits::is_specialized && value_limits::is_bounded);
> ```

I like the direction, but this doesn't work as I intended since with this definition, min/max return `ValueType`, not the `tagged_tagged_integer` type.

Reply:
sorry for the confusion. i didn't suggest remove `min()` or `max()`. i meant to reference `is_specialized = true` when copying this comment from another commit. it is
```c++
static constexpr bool is_specialized = true;
```
which is not necessary, and should be removed or replaced.

Reply:
NP.  Done this way in v2